x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/common/CartesianPoint.java

```diff
@@ -36,18 +37,18 @@
     private static final ParseField Y_FIELD = new ParseField("y");
     private static final ParseField Z_FIELD = new ParseField("z");
 
-    protected float x;
-    protected float y;
+    protected double x;
```

Comment:
Without this change, coordinates could change unexpectedly in a roundtrip like WKT -> `CartesianPoint` -> WKT. It seems common to represent points using doubles even though they're indexed as floats -- for example the `Point` geometry uses doubles.

Reply:
good `Point`. I do not see a problem with this. maybe @nknize has an opinion here